### ISSUE TITLE
Illegal Character in XML output

### ISSUE THREAD

**Issue by Alphastrick** [doxygen/doxygen#7396]
During processing I found the illegal character '&' in a XML attribute.

Example:
```
<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.16">
  <compounddef id="some_compounddef" kind="file" language="lang">
    <!-- ... -->
    <programlisting>
<codeline lineno="1"></codeline>
    </programlisting>
    <location file="path/file_&_name.dox"/>
  </compounddef>
</doxygen>
```

**Comment by albert-github** [comment]
To be able to find the problem just an output snippet is not enough, looks like a problem with an unusual character in a file name.

- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
- Please also specify the doxygen version used.

**Comment by albert-github** [comment]
With some effort I was able to reproduce the problem.
I've just pushed a proposed patch, pull request #7397

**Comment by Alphastrick** [comment]
The used Version of Doxygen is 1.8.16.
Here is the example code:
[issue.zip](https://github.com/doxygen/doxygen/files/3841323/issue.zip)

**Comment by albert-github** [comment]
Please don't closed the issue:
- #7397 is just a proposed patch
- when the proposed patch is integrated also don't close the issue as this will be done at the moment of a release

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
